src/explorer/SiteTreeItem.ts

```diff
@@ -167,6 +178,27 @@ export abstract class SiteTreeItem implements IAzureParentTreeItem {
             workspace.getConfiguration(constants.extensionPrefix, Uri.file(fsPath)).update(constants.configurationSettings.showBuildDuringDeployPrompt, false);
         }
     }
+
+    private async promptToSaveDeployConfigs(node: IAzureNode<SiteTreeItem>, activeWorkspacePath: string, deployedProjectPath: string, telemetryProperties: TelemetryProperties): Promise<void> {
+        const saveDeploymentConfig: string = `Always deploy the workspace "${path.basename(activeWorkspacePath)}" to "${node.treeItem.client.fullName}"?`;
+        const neverSaveDeploymentConfiguration: MessageItem = { title: "Never" };
+        let result: MessageItem = DialogResponses.learnMore;
+        result = await ext.ui.showWarningMessage(saveDeploymentConfig, DialogResponses.yes, neverSaveDeploymentConfiguration, DialogResponses.skipForNow);
+        if (result === DialogResponses.yes) {
+            workspace.getConfiguration(constants.extensionPrefix, Uri.file(deployedProjectPath))
+                .update(constants.configurationSettings.deployToWebAppId, node.id);
+            workspace.getConfiguration(constants.extensionPrefix, Uri.file(deployedProjectPath))
+                .update(constants.configurationSettings.deploySubpath, path.relative(activeWorkspacePath, deployedProjectPath)); // '' is a falsey value
+            telemetryProperties.promptToSaveDeployConfigs = 'Yes';
+        } else if (result === neverSaveDeploymentConfiguration) {
+            workspace.getConfiguration(constants.extensionPrefix, Uri.file(deployedProjectPath))
+                .update(constants.configurationSettings.deployToWebAppId, constants.neverSaveDeploymentConfiguration);
+            telemetryProperties.promptToSaveDeployConfigs = 'Never';
+        }
+        if (!telemetryProperties.promptToSaveDeployConfigs) {
```

Comment:
You did this again - why not just make this an 'else' case for the above if statements?